model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/beam_runner_api.proto

```diff
@@ -965,8 +968,8 @@ message StandardCoders {
     //         01 - on time
     //         10 - late
     //         11 - unknown
-    //     * bit 6 is 1 if this is the first pane, 0 otherwise.
-    //     * bit 7 is 1 if this is the last pane, 0 otherwise.
+    //     * bit 6 is 1 if this is the last pane, 0 otherwise.
```

Comment:
Am I just counting bits from the wrong end? https://github.com/apache/beam/blob/0230b5d420f77c0ea673c36666911e37d1aaa70a/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/PaneInfo.java#L133

Reply:
OK I've read down and see that the Go implementation matches the Java implementation (obvs it would have to) so maybe the best way to express this is to name these by  `byte | 0x01` and `byte | 0x02`

Reply:
Nope, but if you look at the Java and Python linkes (and the internal Dataflow implementation) the updated text is correct. The IsFirst bit is the last bit (index 7) in big endian order.

0b01234567 - indices
0b00000001 - IsFirst bit is true (0x01)
0b00000010 - IsLast bit is true (0x02)

I agree that the original text is more intuitive, but we can't change Python, Java, and Dataflow. It's a small blessing that the Go SDK was the only one impacted by this, and it turns out you can't even use panes properly there either due to https://github.com/apache/beam/issues/31153.

Reply:
Weird, it didn't show my your second comment. Both. The answer is both though. No sense changing from "we have bit indexes" to "they don't mean anything".